src/sql/platform/connection/common/connectionProfile.ts

```diff
@@ -215,14 +215,11 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 	}
 
 	public static fromIConnectionProfile(capabilitiesService: ICapabilitiesService, profile: azdata.IConnectionProfile) {
-		if (profile) {
-			if (profile instanceof ConnectionProfile) {
-				return profile;
-			} else {
-				return new ConnectionProfile(capabilitiesService, profile);
-			}
+		if (profile instanceof ConnectionProfile) {
+			return profile;
+		} else {
+			return new ConnectionProfile(capabilitiesService, profile);
 		}
```

Comment:
You're changing the behavior here - is that intended? I don't really see a huge problem with this change but let's make sure nothing is depending on this functionality

Reply:
Yes, it simplifies the actual change I have to make to have it meet strict null. I checked the callers of this function (and others like this) and verified they aren't just randomly passing in undefined.